rtcdatachannel.go

```diff
@@ -88,27 +92,35 @@ type RTCDataChannel struct {
 	// Onmessage designates an event handler which is invoked on a message
 	// arrival over the sctp transport from a remote peer.
 	//
-	// Deprecated: Variable name is changing from Onmessage to OnMessage.
+	// Deprecated: use OnMessage instead.
 	Onmessage func(datachannel.Payload)
 
+	// OnMessage designates an event handler which is invoked on a message
+	// arrival over the sctp transport from a remote peer.
+	OnMessage func(datachannel.Payload)
+
+	// Deprecated: Will be removed when networkManager is deprecated.
 	rtcPeerConnection *RTCPeerConnection
 }
 
-// // Close closes the RTCDataChannel. It may be called regardless of whether the
-// // RTCDataChannel object was created by this peer or the remote peer.
-// func (d *RTCDataChannel) Close() error {
-// 	if d.ReadyState == RTCDataChannelStateClosing || d.ReadyState == RTCDataChannelStateClosed {
-// 		return nil
-// 	}
-//
-// 	d.ReadyState = RTCDataChannelStateClosing
-//
-// 	// if err := d.rtcPeerConnection.networkManager.SendOpenChannelMessage(d.ID, d.Label); err != nil {
-// 	// 	return &rtcerr.UnknownError{Err: err}
-// 	// }
-// 	return nil
-//
-// }
+func (d *RTCDataChannel) generateID() error {
```

Comment:
IMO it is confusing to have this one the RTCDataChannel itself. It doesn't depend on anything state of the DataChannel we are dispatching off of, and would cause weird behavior if called multiple times.

Reply:
I know... The only reason it is like that is because I was in the middle of transitioning it into the `DataChannel` when I realized that I actually broke master. I will move it for this PR but ultimately it does not belong to `RTCPeerConnection`. It is purely an internal mechanism for leasing the `datachannels` which is being moved into the `sctpTransport` as well.